sn_client/src/chunk_operator.rs

```diff
@@ -0,0 +1,160 @@
+use bytes::Bytes;
+use self_encryption::MIN_ENCRYPTABLE_BYTES;
+use sn_protocol::storage::{Chunk, ChunkAddress};
+use std::fs::File;
+use std::io::{Read, Write};
+use std::path::{Path, PathBuf};
+use xor_name::XorName;
+
+/// Tries to chunk the file, returning
+/// `(head_address, data_map_chunk, file_size, chunk_names)`
+/// and writes encrypted chunks to disk.
+#[derive(Debug, PartialEq)]
+pub struct ChunkOperator {
+    pub head_chunk_address: ChunkAddress,
+    pub data_map: Chunk,
+    pub file_size: u64,
+    pub chunk_vec: Vec<(XorName, PathBuf)>,
+}
+
+#[derive(Default, Clone)]
+pub struct BuildChunkOperator {
+    file_path: PathBuf,
+    chunk_dir: PathBuf,
+    file_size: u64,
+    data_map_chunk: Chunk,
+    chunk_vec: Vec<(XorName, PathBuf)>,
+    // data_map_in_chunks: Vec<(XorName, PathBuf)>,
+}
+
+impl BuildChunkOperator {
+    pub fn new() -> BuildChunkOperator {
+        BuildChunkOperator {
+            file_path: Default::default(),
+            chunk_dir: Default::default(),
+            file_size: 0,
+            data_map_chunk: Default::default(),
+            chunk_vec: vec![],
+        }
+    }
+
+    fn append_data_map(&mut self) -> Vec<(XorName, PathBuf)> {
+        let mut chunks_paths = self.chunk_vec.clone();
+        let data_map_path = self
+            .chunk_dir
+            .join(hex::encode(*self.data_map_chunk.name()));
+        let mut output_file =
+            File::create(data_map_path.clone()).expect("error creating output file");
+
+        // debug!("include_data_map_in_chunks {include_data_map_in_chunks:?}");
+        info!("Data_map_chunk to be written!");
+        trace!("Data_map_chunk being written to {data_map_path:?}");
+
+        output_file
+            .write_all(&self.data_map_chunk.value)
+            .expect("error writing all");
```

Comment:
We might have to return proper errors here. And cannot use `expect` which can panic.